lib/segment/src/types.rs

```diff
@@ -523,6 +524,16 @@ impl From<Map<String, Value>> for Payload {
     }
 }
 
+pub fn serialize_payload<S>(point: &Option<Payload>, serializer: S) -> Result<S::Ok, S::Error>
+where
+    S: serde::Serializer,
+{
+    point
+        .as_ref()
+        .filter(|payload| !payload.is_empty())
+        .serialize(serializer)
+}
```

Comment:
Disagreed.
Payload should not be null if user uploaded `{}` as payload. 
Overall, I think, payload should only be `null` if it is explicitly disabled in request: `"with_payload": false`

Reply:
Agree. I was trying to "cheat" this `{}` vs. `null` errors, but then realized it's a dumb idea. I'll fix this properly tomorrow.

However, now that you mentioned that user can upload `{}` explicitly... we do not distinguish between "no payload" and "empty payload" in the gRPC API (which means we do not distinguish them in the internal API as well), so how can we tell one from the other? 🤔

Seems like the best we can do is to ensure we return `null` if `with_payload: false`, and `{}` if `with_payload: true`.

Reply:
It's impossible to add an `optional` attribute to a `map` field in ProtoBuf, so I had to change `Record` and `ScoredPoint` conversions from their corresponding proto-messages.